scripts/ci/check_compliance.py

```diff
@@ -397,6 +399,9 @@ def parse_kconfig(self):
         except kconfiglib.KconfigError as e:
             self.failure(str(e))
             raise EndTest
+        finally:
+            # Clean up the temporary directory
+            shutil.rmtree(kconfiglib_dir)
```

Comment:
This may be unnecessary, since the file should be automatically removed with garbage collection.

https://docs.python.org/3/library/tempfile.html

Reply:
Is that applicable for `mkdtemp` as well? Or I could just use `TemporaryDirectory` if not?

Reply:
oh and looks like this was only added in 3.10.